Documentation/network/servicemesh/mutual-authentication/mutual-authentication.rst

```diff
@@ -0,0 +1,116 @@
+
+Unlike traditional TLS, which involves a one-way authentication process where the client verifies the server's identity, mutual TLS adds an additional layer of security by requiring both the client and the server to authenticate each other.
+
+Mutual TLS is also known as two-way TLS. Let's remind ourselves the goals behind TLS.
+
+TLS Goals
+---------
+
+`RFC-8446 <https://datatracker.ietf.org/doc/html/rfc8446#section-1>`_ defines the goals of a TLS secure channel as having the following properties:
+
+- Authentication: The server side of the channel is always authenticated; the client side is optionally authenticated.
+  Authentication can happen via asymmetric cryptography (e.g., RSA [RSA], the Elliptic Curve Digital Signature Algorithm (ECDSA)
+  [ECDSA], or the Edwards-Curve Digital Signature Algorithm (EdDSA) [RFC8032]) or a symmetric pre-shared key (PSK).
+
+- Confidentiality: Data sent over the channel after establishment is only visible to the endpoints. TLS does not hide the length of
+  the data it transmits, though endpoints are able to pad TLS records in order to obscure lengths and improve protection against
+  traffic analysis techniques.
+
+- Integrity: Data sent over the channel after establishment cannot be modified by attackers without detection.
+
+In traditional TLS transactions, these are handled by having a public key handshake performed (which satisfies the *Authentication* requirement), which then negotiates a symmetric key that is used by a record protocol to encrypt the traffic in transit (which satisfies the *Confidentiality* and *Integrity* properties). 
+
+MTLS in Cilium
+##############
+
+Cilium’s solution splits the handshake protocol and record protocol apart and performs them out of band of the actual packet flow.
+
+In Cilium’s case, packets are dropped until the authentication has been completed, but once the handshake is completed, other handshakes are not required until the certificate expires (however, the implementation will re-authenticate when new identities are issued, without dropping packets if it’s before expiry).
+
+The Integrity and Confidentiality properties are maintained using Cilium’s existing encryption mechanisms, :ref:`encryption_wg` and :ref:`encryption_ipsec`, which can be enabled to automatically create and maintain encrypted connections between pods.
+
+In short, Cilium’s new Mutual Authentication support brings the mutual authentication handshake out-of-band for regular connections, and uses the existing traffic encryption mechanisms already in place to meet the encryption and confidentiality requirements of TLS.
+
+To address the challenge of identity verification in dynamic and heterogeneous environments, mutual authentication requires a framework secure identity verification for distributed systems.
+
+Identity Management
```

Comment:
I think this section is fine, it's needed background for the walkthrough stuff later. We should make sure we link to the SPIFFE and SPIRE sites as well.